xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/viewers/attachments.js

```diff
@@ -216,10 +216,13 @@ require(['jquery', 'xwiki-events-bridge'], function($) {
         notification = new XWiki.widgets.Notification(l10n['core.viewers.attachments.delete.inProgress'], 'inprogress');
       },
       success : function() {
-        window[thisLivetableName].refresh();
-        if (thisLivetableName === 'docAttachments') {
-          updateCount();
-        }
+        liveData.updateEntries().then(() => {
+          const liveDataId = liveData.data.id;
+          liveData = $('#' + liveDataId).data('liveData');
```

Comment:
Is this really needed? I mean, the root element holding the reference to the live data instance shouldn't change when the list of entries is updated, and neither the live data instance (it should be the same instance).